pkg/reconciler/reconciler.go

```diff
@@ -567,8 +571,11 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (res ctrl.
 	u.UpdateStatus(updater.EnsureCondition(conditions.Initialized(corev1.ConditionTrue, "", "")))
 
 	if obj.GetDeletionTimestamp() != nil {
-		err := r.handleDeletion(ctx, actionClient, obj, log)
-		return ctrl.Result{}, err
+		if err := r.handleDeletion(ctx, actionClient, obj, log); err != nil {
+			return ctrl.Result{}, err
```

Comment:
Just to make sure I'm following along - this is a case in which we would still want the deferred update function to occur to make sure we are updating any statuses with failures?

Reply:
yes :)